pypga/modules/migen/spi.py

```diff
@@ -24,65 +24,45 @@ def __init__(
         clock_polarity: bool = False,  # clock polarity
         clock_phase: bool = False,  # clock phase
         lsb_first: bool = False,  # send LSB rather than MSB first
+        leading_halfcycles: Union[Signal, int] = 1,
+        trailing_halfcycles: Union[Signal, int] = 1,
+        idle_halfcycles: Union[Signal, int] = 1,
     ):
         if data_width <= 16:
             raise ValueError("This module does not support data_width < 17 bits.")
-        # signals to connect to SPI wires
+        # Signals to connect to SPI wires
         self.mosi = Signal()
-        self.miso = Signal()
+        # Currently, MISO is not implemented
```

Comment:
I have removed MISO for now, since we don't need it at the moment and also cannot properly test it. For a full-fledged SPI interface, of course it should be there.